### DISCUSSION TITLE
Need help integrating large language models like ChatGPT to write code / content into nonprofit slides

### DISCUSSION THREAD

**Discussion by jaanaltosaar** [slidevjs/slidev#1615] [Q&A]
**Is your feature request related to a problem? Please describe.**

I need to demo our approach at @onefact to leveraging large language models like ChatGPT for non-profit use cases (I'm giving a talk in a few days).

Here are the outputs I need to demo:

https://colab.research.google.com/github/jaanli/language-model-notebooks/blob/main/notebooks/getting-started.ipynb

https://colab.research.google.com/github/onefact/electronic-health-records/blob/main/notebooks/loading_physionet_mimic_data.ipynb

https://onefact.github.io/synthetic-healthcare-data/

https://jaanli.github.io/new-york-real-estate/

https://jaanli.github.io/american-community-survey/new-york-area/income-by-race

**Describe the solution you'd like**
Very little of the above code is written by our team. It is mainly written by claude.ai. 

We need a way to communicate this concept to the audiences we are presenting to over the summer, including government entities and investors.

It is difficult to make an API call from Powerpoint or Keynote, so I thought I should try something based on `vite` because of how flexible and fast it is (these LLM APIs are the opposite of fast...).

**Describe alternatives you've considered**
Having a screen recording is the worst case scenario (then people looking at our non-profit website after will need to see youtube ads, or other ways of hosting it - we use GitHub pages).

**Comment by kermanx** [comment]
Do you mean embedding the page, or want a real-time output from LLM in the slides?

**Reply by jaanaltosaar** [reply]
real-time output from the LLM, like in these examples: 

Simple: https://colab.research.google.com/github/jaanli/language-model-notebooks/blob/main/notebooks/getting-started.ipynb

Complex:

https://colab.research.google.com/github/onefact/electronic-health-records/blob/main/notebooks/loading_physionet_mimic_data.ipynb 

(I didn't write any code for these, the LLM just fills it in the code cells so am hoping to do a live demo of either approach!)

**Comment by jaanaltosaar** [comment]
Perhaps embedding https://github.com/jupyterlite/jupyterlite could be a solution here? Not sure how well slidev interacts with Jupyter.

**Reply by kermanx** [reply]
Slidev doesn't have built-in support for Jupyter yet. However, we can embed the Jupyter notebook via an `iframe`

**Reply by jaanaltosaar** [reply]
That's a great idea, thank you!!!!
